DPM-248 Try get ccache from sun.security.krb5 is KRB5CCNAME not set

BenoitHanotte opened the pull request:
In case KRB5CCNAME is not set, presto might still be able to get the
ticket cache location from the sun.security.krb5 implementation.

This allows clients on MacOS or on linux without the environment variable
set to still connect to the cluster without having to provide it
manually.

piyushnarang commented:
@szehonCriteo - please take a look when you get a chance. Would be nice to have this covered in some unit tests but it looks like we'll need to mock a bunch of stuff for that.

Commit:
Get ccache from sun.security.krb5 if KRB5CCNAME not set

DPM-248
In case KRB5CCNAME is not set, we might still be able to get the ticket
cache location using the Sun APIs.
This will allow unix clients (including MacOS) to connect despite this
env variable not being set.

EricMi reviewed (approved):
LGTM
